clients/src/main/java/org/apache/kafka/common/requests/HeartbeatRequest.java

```diff
@@ -37,6 +38,10 @@ public Builder(HeartbeatRequestData data) {
 
         @Override
         public HeartbeatRequest build(short version) {
+            if (data.groupInstanceId() != null && version < 3) {
+                throw new UnsupportedVersionException("The broker join group protocol version " + version + " doesn't support static membership." +
```

Comment:
nit: missing space at the end of the sentence

Same thing in the other builders.